code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_ai.dm

```diff
@@ -0,0 +1,310 @@
+	var/list/targets2 = list()
+	var/list/targets3 = list()
+	for(var/mob/living/H in view(src, vision_range))
+		if(H.stat == DEAD)
+			continue
+		if(H.flags & GODMODE)
+			continue
+		if(H.stat == UNCONSCIOUS && !stat_attack)
+			continue
+		if(ai_type == TS_AI_DEFENSIVE && !(H in enemies))
+			continue
+		if(isterrorspider(H))
+			if(H in enemies)
+				targets3 += H
+			continue
+		if(iscarbon(H))
+			var/mob/living/carbon/C = H
+			if(IsInfected(C)) // only target the infected if they're attacking us. Even then, lowest priority.
+				if(C in enemies)
+					targets3 += C
+					continue
+			else if(C.reagents.has_reagent("terror_black_toxin",60))
+				// only target those dying of black spider venom if they are close, or our enemy
+				if(get_dist(src,C) <= 2 || (C in enemies))
+					targets2 += C
+			else
+				// Target prioritization by spider type. BRUTE spiders prioritize lower armor values, POISON spiders prioritize poisonable targets
+				if(ai_target_method == TS_DAMAGE_BRUTE)
+					var/theirarmor = C.getarmor(type = "melee")
+					// Example values: Civilian: 2, Engineer w/ Hardsuit: 10, Sec Officer with armor: 19, HoS: 48, Deathsquad: 80
+					if(theirarmor < 10)
+						targets1 += C
+					else if(C in enemies)
+						if(theirarmor < 30)
+							targets2 += C
+						else
+							targets3 += C
+					else
+						targets3 += C
+				else if(ai_target_method == TS_DAMAGE_POISON)
+					if(C.can_inject(null,0,"chest",0))
+						targets1 += C
+					else if(C in enemies)
+						targets2 += C
+					else
+						targets3 += C
+				else
+					// TS_DAMAGE_SIMPLE
+					if(C in enemies)
+						targets2 += C
+					else
+						targets3 += C
+		else
+			if(istype(H,/mob/living/simple_animal))
+				var/mob/living/simple_animal/S = H
+				if(S.force_threshold > melee_damage_upper)
+					continue
+			if(H in enemies)
+				targets2 += H
+			else
+				targets3 += H
+	if(health < maxHealth)
+		// if we're hurt, and ONLY if we're hurt, do the additional check for mechs/space pods. Save processing time.
+		for(var/obj/mecha/M in view(src, vision_range))
+			if(get_dist(M, src) <= 2)
+				targets2 += M
+			else
+				targets3 += M
+		for(var/obj/spacepod/S in view(src, vision_range))
+			targets3 += S
+	if(targets1.len)
+		return targets1
+	if(targets2.len)
+		return targets2
+	return targets3
+
+/mob/living/simple_animal/hostile/poison/terror_spider/LoseTarget()
+	if(target && isliving(target))
+		var/mob/living/T = target
+		if(T.stat > 0)
+			killcount++
+			regen_points += regen_points_per_kill
+	attackstep = 0
+	attackcycles = 0
+	..()
+
+// --------------------------------------------------------------------------------
+// --------------------- TERROR SPIDERS: AI BEHAVIOR CODE -------------------------
+// --------------------------------------------------------------------------------
+
+/mob/living/simple_animal/hostile/poison/terror_spider/handle_automated_action()
+	if(!stat && !ckey) // if we are not dead, and we're not player controlled
```

Comment:
It would be preferable that you format this as such
```
if(stat || ckey)
    return

// other code here
```
In order to avoid having the major block of code tabbed out two to the right.